src/System.CommandLine.Tests/ParserTests.cs

```diff
@@ -1874,6 +1874,28 @@ public void Location_offset_correct_when_colon_or_equal_used()
             result2.Locations.Single().Should().Be(expectedLocation2);
         }
 
+        [Fact]
+        public void Locations_correct_for_collection()
+        {
+            var option1 = new CliOption<string[]>("--opt1");
+            option1.AllowMultipleArgumentsPerToken = true;
+            var rootCommand = new CliRootCommand
+                    {
+                        option1
+                    };
+            var expectedOuterLocation = new Location("testhost", Location.User, -1, null);
+            var expectedLocation1 = new Location("Kirk", Location.User, 2, expectedOuterLocation);
+            var expectedLocation2 = new Location("Spock", Location.User, 3, expectedOuterLocation);
+            var expectedLocation3 = new Location("Uhura", Location.User, 4, expectedOuterLocation);
+
+            var parseResult = CliParser.Parse(rootCommand, "subcommand --opt1 Kirk Spock Uhura");
+
+            var commandResult = parseResult.CommandResult;
+            parseResult.GetValue(option1);
+            var result1 = parseResult.GetValueResult(option1);
+            result1.Locations.Should().BeEquivalentTo([expectedLocation1, expectedLocation2, expectedLocation3]);
+        }
```

Comment:
```suggestion
            var option = new CliOption<string[]>("--opt1");
            option.AllowMultipleArgumentsPerToken = true;
            var rootCommand = new CliRootCommand
                    {
                        option
                    };
            var expectedOuterLocation = new Location("testhost", Location.User, -1, null);
            var expectedLocation1 = new Location("Kirk", Location.User, 2, expectedOuterLocation);
            var expectedLocation2 = new Location("Spock", Location.User, 3, expectedOuterLocation);
            var expectedLocation3 = new Location("Uhura", Location.User, 4, expectedOuterLocation);

            var parseResult = CliParser.Parse(rootCommand, "subcommand --opt1 Kirk Spock Uhura");

            var commandResult = parseResult.CommandResult;
            parseResult.GetValue(option);
            var result = parseResult.GetValueResult(option);
            result.Locations.Should().BeEquivalentTo([expectedLocation1, expectedLocation2, expectedLocation3]);
```
Changing `option1` to `option` and `result1` to `result`.

- Having `expectedLocation1` (and 2 and 3) have index's of 2,3,4 is a bit weird for some reason, is there a reason for this? Just trying to understand what is happening here.
- Is the assignment to `commandResult` for just debugging? And another onion issue here for debugging as with the `valueSubSystemTests`?